<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Peyton Chiu">

<title>Sequence and Disorder Features Underlying Functional Domains of p53</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Report_files/libs/clipboard/clipboard.min.js"></script>
<script src="Report_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="Report_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="Report_files/libs/quarto-html/popper.min.js"></script>
<script src="Report_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Report_files/libs/quarto-html/anchor.min.js"></script>
<link href="Report_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Report_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Report_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Report_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Report_files/libs/bootstrap/bootstrap-bb462d781dde1847d9e3ccf7736099dd.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Sequence and Disorder Features Underlying Functional Domains of p53</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Peyton Chiu </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="section" class="level2">
<h2 class="anchored" data-anchor-id="section"></h2>
</section>
<section id="setting-up-the-analysis-environment" class="level2">
<h2 class="anchored" data-anchor-id="setting-up-the-analysis-environment">Setting up the analysis environment</h2>
<p>Before starting our analysis, we need to properly configure our work environment. For this project, we’ll be using R, in particular the tidyverse and Biostrings packages.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>With our libraries now loaded in we can load in the fasta file for p53 in humans. For this project the file was taken from Uniprot (<strong>P04637)</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>p53_seq <span class="ot">&lt;-</span> <span class="fu">readAAStringSet</span>(<span class="st">"C:/Users/USER/Desktop/P53_protein_proj/data/P53_human.fasta"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Before proceeding it might be a good idea to convert our data into a table</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>seq_str <span class="ot">&lt;-</span> <span class="fu">as.character</span>(p53_seq[[<span class="dv">1</span>]])</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>aa_vec <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(seq_str, <span class="st">""</span>))</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>p53_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">RESID =</span> <span class="fu">seq_along</span>(aa_vec),</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">AminoAcid =</span> aa_vec</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Before proceeding we should a quick sanity check on the contents</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(p53_table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     RESID      AminoAcid        
 Min.   :  1   Length:393        
 1st Qu.: 99   Class :character  
 Median :197   Mode  :character  
 Mean   :197                     
 3rd Qu.:295                     
 Max.   :393                     </code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(p53_table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  RESID AminoAcid
1     1         M
2     2         E
3     3         E
4     4         P
5     5         Q
6     6         S</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(p53_table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    RESID AminoAcid
388   388         E
389   389         G
390   390         P
391   391         D
392   392         S
393   393         D</code></pre>
</div>
</div>
<p>We first examine the intrinsic disorder scores, which quantify the likelihood that a given amino acid residue resides within an intrinsically disordered region (IDR). For this project, disorder scores are interpreted according to the following convention, adapted from the referenced source: <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC7279152/#:~:text=For%20each%20protein%2C%20per%2Dresidue,corresponding%20data%20are%20outlined%20below.">here</a></p>
<blockquote class="blockquote">
<ul>
<li><p>Scores &lt; 0.5: The residue is generally ordered.</p></li>
<li><p>Scores &gt; 0.5: The residue is generally considered disordered (unstructured or flexible).</p></li>
<li><p>Scores between 0.25 and 0.5: These residues are often considered flexible or moderately disordered</p></li>
</ul>
</blockquote>
<p>We can get our scores using the <a href="https://iupred2a.elte.hu/">IUPred2A</a> site, and we will be only utilizing the IUPred2 values</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(jsonlite)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'jsonlite' was built under R version 4.5.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'jsonlite'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    flatten</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>dis_scores <span class="ot">&lt;-</span> <span class="fu">fromJSON</span>(<span class="st">"C:/Users/USER/Desktop/P53_protein_proj/data/p53_iupred2a.json"</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>p53_table<span class="sc">$</span>Disorder_Scores <span class="ot">&lt;-</span>dis_scores<span class="sc">$</span>iupred2</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(p53_table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "RESID"           "AminoAcid"       "Disorder_Scores"</code></pre>
</div>
</div>
<p>Now that the scores are added into the table, it might be a good idea to get a quick visualization of the overall values of our disorder scores through ggplot</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(p53_table, <span class="fu">aes</span>(<span class="at">x =</span> RESID, <span class="at">y =</span> Disorder_Scores)) <span class="sc">+</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Intrinsic Disorder Profile of Human p53"</span>,</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Residue Number"</span>,</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"IUPred2A Disorder Score"</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Report_files/figure-html/p53disorder_domain_unmarked.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This profile of disorder scores aligns with what we should expect from p53 protein. Both the C and N termini display consistently high scores, which is to be expected as both regions require a high degree of flexibility to carry out interactions with multiple binding partners.</p>
</section>
<section id="mapping-out-the-individual-domains-per-residue" class="level2">
<h2 class="anchored" data-anchor-id="mapping-out-the-individual-domains-per-residue">Mapping out the Individual Domains per Residue</h2>
<p>With our disorder scores calculated, we should now organize them into distinct domains. For this project, I utilized the domain models as shown in this article <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC3905691/?utm_source=chatgpt.com">here</a></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>p53_table<span class="sc">$</span>Domain <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(p53_table)) {</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  r <span class="ot">&lt;-</span> p53_table<span class="sc">$</span>RESID[i]   </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (r <span class="sc">&lt;=</span> <span class="dv">40</span>) {</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>    p53_table<span class="sc">$</span>Domain[i] <span class="ot">&lt;-</span> <span class="st">"TAD1"</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (r <span class="sc">&gt;</span> <span class="dv">40</span> <span class="sc">&amp;&amp;</span> r <span class="sc">&lt;=</span> <span class="dv">60</span>) {</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>    p53_table<span class="sc">$</span>Domain[i] <span class="ot">&lt;-</span> <span class="st">"TAD2"</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (r <span class="sc">&gt;=</span> <span class="dv">64</span> <span class="sc">&amp;&amp;</span> r <span class="sc">&lt;=</span> <span class="dv">92</span>) {</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>    p53_table<span class="sc">$</span>Domain[i] <span class="ot">&lt;-</span> <span class="st">"PR"</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (r <span class="sc">&gt;=</span> <span class="dv">94</span> <span class="sc">&amp;&amp;</span> r <span class="sc">&lt;=</span> <span class="dv">292</span>) {</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>    p53_table<span class="sc">$</span>Domain[i] <span class="ot">&lt;-</span> <span class="st">"DBD"</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (r <span class="sc">&gt;=</span> <span class="dv">325</span> <span class="sc">&amp;&amp;</span> r <span class="sc">&lt;=</span> <span class="dv">356</span>) {</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>    p53_table<span class="sc">$</span>Domain[i] <span class="ot">&lt;-</span> <span class="st">"OD"</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (r <span class="sc">&gt;</span> <span class="dv">356</span> <span class="sc">&amp;&amp;</span> r <span class="sc">&lt;=</span> <span class="dv">393</span>) {</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>    p53_table<span class="sc">$</span>Domain[i] <span class="ot">&lt;-</span> <span class="st">"CTD"</span></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>    p53_table<span class="sc">$</span>Domain[i] <span class="ot">&lt;-</span> <span class="st">"Other"</span>  </span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>After assigning each residue to a domain, we should do a quick check over our results</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(p53_table<span class="sc">$</span>Domain)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "TAD1" "TAD1" "TAD1" "TAD1" "TAD1" "TAD1"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(p53_table<span class="sc">$</span>Domain)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CTD" "CTD" "CTD" "CTD" "CTD" "CTD"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(p53_table<span class="sc">$</span>Domain)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
  CTD   DBD    OD Other    PR  TAD1  TAD2 
   37   199    32    36    29    40    20 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p53_table<span class="sc">$</span>Domain[<span class="dv">90</span><span class="sc">:</span><span class="dv">100</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "PR"    "PR"    "PR"    "Other" "DBD"   "DBD"   "DBD"   "DBD"   "DBD"  
[10] "DBD"   "DBD"  </code></pre>
</div>
</div>
</section>
<section id="calculating-for-kd-scores-and-charge" class="level2">
<h2 class="anchored" data-anchor-id="calculating-for-kd-scores-and-charge">Calculating for KD scores and Charge</h2>
<p>Proteins function through local sequence regions rather than isolated residues; therefore, rolling averages of hydrophobicity and charge are more appropriate in illustrating the regional chemical environments. For this section we will be using the zoo and dplyr libraries (the latter has been added for easier modification of our existing data table)</p>
<p>To first get the rolling averages, we need to input our raw data. For this project, I’ve used the KD scores as shown <a href="https://chem.libretexts.org/Courses/CSU_Chico/CSU_Chico%3A_CHEM_451_-_Biochemistry_I/CHEM_451_Test/04%3A_Protein_Structure/4.7%3A_G._Predicting_Protein_Properties_From_Sequences/G3._Prediction_of_Hydrophobicity?utm_source=chatgpt.com">here</a></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># assigning the KD values</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>KD_val <span class="ot">&lt;-</span> <span class="fu">c</span>( <span class="at">A =</span> <span class="fl">1.8</span>, <span class="at">R =</span> <span class="sc">-</span><span class="fl">4.5</span>, <span class="at">N =</span> <span class="sc">-</span><span class="fl">3.5</span>, <span class="at">D =</span> <span class="sc">-</span><span class="fl">3.5</span>, <span class="at">C =</span> <span class="fl">2.5</span>, <span class="at">Q =</span> <span class="sc">-</span><span class="fl">3.5</span>, <span class="at">E =</span> <span class="sc">-</span><span class="fl">3.5</span>, <span class="at">G =</span> <span class="sc">-</span><span class="fl">0.4</span>, <span class="at">H =</span> <span class="sc">-</span><span class="fl">3.2</span>, <span class="at">I =</span> <span class="fl">4.5</span>, <span class="at">L =</span> <span class="fl">3.8</span>, <span class="at">K =</span> <span class="sc">-</span><span class="fl">3.9</span>, <span class="at">M =</span> <span class="fl">1.9</span>, <span class="at">F =</span> <span class="fl">2.8</span>, <span class="at">P =</span> <span class="sc">-</span><span class="fl">1.6</span>, <span class="at">S =</span> <span class="sc">-</span><span class="fl">0.8</span>, <span class="at">T =</span> <span class="sc">-</span><span class="fl">0.7</span>, <span class="at">W =</span> <span class="sc">-</span><span class="fl">0.9</span>, <span class="at">Y =</span> <span class="sc">-</span><span class="fl">1.3</span>, <span class="at">V =</span> <span class="fl">4.2</span> ) </span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>p53_table<span class="sc">$</span>KD <span class="ot">&lt;-</span> KD_val[p53_table<span class="sc">$</span>AminoAcid]</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="co"># assigning the charge values</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>charge_val <span class="ot">&lt;-</span> <span class="fu">c</span>( <span class="at">K =</span><span class="dv">1</span>, <span class="at">R =</span><span class="dv">1</span> , <span class="at">D=</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">E =</span><span class="sc">-</span><span class="dv">1</span>) </span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>p53_table<span class="sc">$</span>Charge <span class="ot">&lt;-</span> charge_val[p53_table<span class="sc">$</span>AminoAcid]</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> ( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(p53_table)){</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>  c <span class="ot">&lt;-</span> p53_table<span class="sc">$</span>Charge[i]</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.na</span>(c)){</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>    p53_table<span class="sc">$</span>Charge[i] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>With our raw values calculated, we can now tabulate the rolling averages. For this, we are going to have our averages be aligned on the center as we are taking basing our values of all residues nearby. A window of 9 was chosen for this calculation, taken from <a href="https://www.dnastar.com/manuals/protean3d/17.2/en/topic/hydropathy-kyte-doolittle?utm_source=chatgpt.com">here</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(zoo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'zoo' was built under R version 4.5.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'zoo'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    as.Date, as.Date.numeric</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>p53_table <span class="ot">&lt;-</span> p53_table <span class="sc">%&gt;%</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">avg_kd =</span> <span class="fu">rollmean</span>(KD, <span class="at">k =</span> <span class="dv">9</span>, <span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">align =</span> <span class="st">"center"</span>),</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">avg_charge =</span> <span class="fu">rollmean</span>(Charge, <span class="at">k =</span> <span class="dv">9</span>, <span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">align =</span> <span class="st">"center"</span>)</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Note:</strong> Since we are using rolling averages with a 9-residue sliding window, we should expect terminal residues lacking a full neighborhood to be left undefined (N/A).</p>
<p>Before proceeding, we should do some quick visualizations via ggplot</p>
<p><strong>For KD and its rolling average</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(p53_table, <span class="fu">aes</span>(<span class="at">x =</span> RESID, <span class="at">y =</span> avg_kd)) <span class="sc">+</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">"turquoise"</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Moving KD (Kyte–Doolittle) averages in p53"</span>,</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Residue Number"</span>,</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"KD (Kyte–Doolittle) Score"</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 8 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Report_files/figure-html/Moving_KD_avg.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>For charge and its rolling average</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(p53_table, <span class="fu">aes</span>(<span class="at">x =</span> RESID, <span class="at">y =</span> avg_charge)) <span class="sc">+</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">"turquoise"</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Moving Charge averages in p53"</span>,</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Residue Number"</span>,</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Charge"</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 8 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Report_files/figure-html/Moving_Charge_avg.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="interpreting-our-data" class="level2">
<h2 class="anchored" data-anchor-id="interpreting-our-data">Interpreting our data</h2>
<p>With our table filled out, we can start to create graphs that show the whole story. Let’s begin by creating a graph of disorder with the domains overlaid.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">start =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">41</span>, <span class="dv">64</span>, <span class="dv">94</span>, <span class="dv">325</span>, <span class="dv">357</span>),</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">end   =</span> <span class="fu">c</span>(<span class="dv">40</span>, <span class="dv">60</span>, <span class="dv">92</span>, <span class="dv">292</span>, <span class="dv">356</span>, <span class="dv">393</span>),</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> <span class="fu">c</span>(<span class="st">"TAD1"</span>, <span class="st">"TAD2"</span>, <span class="st">"PR"</span>, <span class="st">"DBD"</span>, <span class="st">"OD"</span>, <span class="st">"CTD"</span>)</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>midpoint <span class="ot">&lt;-</span> (df<span class="sc">$</span>start <span class="sc">+</span> df<span class="sc">$</span>end) <span class="sc">/</span> <span class="dv">2</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(p53_table, <span class="fu">aes</span>(<span class="at">x =</span> RESID, <span class="at">y =</span> Disorder_Scores)) <span class="sc">+</span></span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">40</span>), <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="dv">41</span>, <span class="dv">60</span>), <span class="at">color =</span> <span class="st">"yellow"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="dv">64</span>, <span class="dv">92</span>), <span class="at">color =</span> <span class="st">"green"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="dv">94</span>, <span class="dv">292</span>), <span class="at">color =</span> <span class="st">"orange"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="dv">325</span>, <span class="dv">356</span>), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="dv">357</span>, <span class="dv">393</span>), <span class="at">color =</span> <span class="st">"turquoise"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(</span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">"text"</span>,</span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> df<span class="sc">$</span>midpoint,</span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="cn">Inf</span>,</span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> df<span class="sc">$</span>label,</span>
<span id="cb36-25"><a href="#cb36-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">vjust =</span><span class="dv">20</span>,</span>
<span id="cb36-26"><a href="#cb36-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb36-27"><a href="#cb36-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span><span class="fl">0.5</span></span>
<span id="cb36-28"><a href="#cb36-28" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb36-29"><a href="#cb36-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-30"><a href="#cb36-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">clip =</span> <span class="st">"off"</span>) <span class="sc">+</span></span>
<span id="cb36-31"><a href="#cb36-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-32"><a href="#cb36-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb36-33"><a href="#cb36-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Intrinsic Disorder Profile of Human p53"</span>,</span>
<span id="cb36-34"><a href="#cb36-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Residue Number"</span>,</span>
<span id="cb36-35"><a href="#cb36-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"IUPred2A Disorder Score"</span></span>
<span id="cb36-36"><a href="#cb36-36" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb36-37"><a href="#cb36-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-38"><a href="#cb36-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Report_files/figure-html/p53disorder_full.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><em>Although this plot is a bit hard to read, it clearly illustrates the relationship between intrinsic disorder and functional domains. The N- and C-terminal regions exhibit the highest disorder levels, whereas the DNA-binding domain shows the lowest disorder.</em></p>
</section>
<section id="domain-level-comparison" class="level2">
<h2 class="anchored" data-anchor-id="domain-level-comparison">Domain Level Comparison</h2>
<p>To get a better assessment of our data we should perform a domain level comparsion to assess whether intrinsic disorder and hydrophobicity differ across functional regions of p53.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co">#hardcode the domain order </span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>p53_table<span class="sc">$</span>Domain <span class="ot">&lt;-</span> <span class="fu">factor</span>(</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  p53_table<span class="sc">$</span>Domain,</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"TAD1"</span>, <span class="st">"TAD2"</span>, <span class="st">"PR"</span>, <span class="st">"DBD"</span>, <span class="st">"OD"</span>, <span class="st">"CTD"</span>)</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>bp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(p53_table, <span class="fu">aes</span>(<span class="at">x=</span>Domain, <span class="at">y=</span>Disorder_Scores, <span class="at">fill=</span>Domain)) <span class="sc">+</span> </span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Disorder score by domain"</span>,<span class="at">x=</span><span class="st">"Domain"</span>, <span class="at">y =</span> <span class="st">"Disorder Score"</span>)</span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>bp <span class="sc">+</span> <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Report_files/figure-html/p53KDviadomain.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Intrinsic disorder varies strongly across p53 domains. The N-terminal domains (TAD1 and TAD2), the proline-rich region, and the C-terminal domain exhibit high disorder scores, consistent with flexible regulatory regions. In contrast, the DNA-binding domain shows substantially lower disorder, indicating a predominantly structured inner core. The oligomerization domain displays slightly higher disorder scores compared to DBD but lower than the rest of the regions. Residues that did not fit into any domain exhibited very high disorder scores, supporting the idea that they serve as flexible linker regions.</p>
<p>In short regions, which tend to be the site for gene regulation and activation, seem to have higher amounts of disorder, while the regions used for the actual binding to DNA have lower.</p>
<p>We should perform the same assessment for our rolling averages Kyle Doolite scores</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>bp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(p53_table, <span class="fu">aes</span>(<span class="at">x=</span>Domain, <span class="at">y=</span>avg_kd, <span class="at">fill=</span>Domain)) <span class="sc">+</span> </span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"KD scores (rolling averages) by domain"</span>,<span class="at">x=</span><span class="st">"Domain"</span>, <span class="at">y =</span> <span class="st">"Kyle Doolite Scores"</span>)</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>bp <span class="sc">+</span> <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 8 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Report_files/figure-html/p53KDviadomain.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Median Kyte–Doolittle hydrophobicity values differed across p53 domains. The DNA-binding domain showed the least negative median value (−0.5), consistent with a structured and stably folded region. In contrast, regulatory and disordered regions exhibited more negative medians, including TAD1 (−0.75), TAD2 (−1.2), and the C-terminal domain (−1.3), indicating increased hydrophilicity. The oligomerization domain occupied an intermediate range (−0.8), while linker regions displayed the most hydrophilic values (−1.6). The proline-rich region showed near-neutral hydrophobicity, consistent with disorder driven by residue composition rather than hydrophobicity alone.</p>
<p>Despite this smoothing, clear hydrophobicity differences remain between structured and disordered domains with patterns mirroring the domain’s overall intrinsic disorder (lower disordered regions exhibited higher hydrophobicity consistent with the idea of structured hydrophobic inner core.</p>
</section>
<section id="validating-our-results" class="level2">
<h2 class="anchored" data-anchor-id="validating-our-results">Validating our results</h2>
<p>While our graphs seem to support the idea of the DBD differing from the rest of regions, we can verify them via statistical tests.</p>
<p>For this project, we’ll be using the Wilcoxon rank-sum test as the data does not have a normal distribution.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Organize the data into DBD regions and non DBD regions</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>p53_table<span class="sc">$</span>DBD <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  p53_table<span class="sc">$</span>Domain <span class="sc">==</span> <span class="st">"DBD"</span>,</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Yes"</span>,</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"No"</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a><span class="co"># vectorizing the values</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a><span class="co"># for the Disorder values </span></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>NonDBD_values <span class="ot">&lt;-</span> p53_table<span class="sc">$</span>Disorder[p53_table<span class="sc">$</span>DBD <span class="sc">==</span> <span class="st">"No"</span>]</span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>DBD_values <span class="ot">&lt;-</span> p53_table<span class="sc">$</span>Disorder[p53_table<span class="sc">$</span>DBD <span class="sc">==</span> <span class="st">"Yes"</span>]</span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a><span class="co"># for the rolling KD averages </span></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>KD_NonDBD_values <span class="ot">&lt;-</span> p53_table<span class="sc">$</span>avg_kd[p53_table<span class="sc">$</span>DBD <span class="sc">==</span> <span class="st">"No"</span>]</span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>KD_DBD_values <span class="ot">&lt;-</span> p53_table<span class="sc">$</span>avg_kd[p53_table<span class="sc">$</span>DBD <span class="sc">==</span> <span class="st">"Yes"</span>]</span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(</span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a>  NonDBD_values,</span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a>  DBD_values</span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  NonDBD_values and DBD_values
W = 29290, p-value &lt; 2.2e-16
alternative hypothesis: true location shift is not equal to 0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>  KD_NonDBD_values,</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>  KD_DBD_values </span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  KD_NonDBD_values and KD_DBD_values
W = 12033, p-value = 0.001942
alternative hypothesis: true location shift is not equal to 0</code></pre>
</div>
</div>
<p>Disorder scores were significantly higher outside the DNA-binding domain than within the DNA-binding domain (p &lt; 2.2 × 10⁻¹⁶).</p>
<p>Kyte Doolittle rolling-average hydrophobicity values also differed significantly between non-DBD regions and the DNA-binding domain (p = 0.0019), with non-DBD regions exhibiting more hydrophilic (lower KD) values.</p>
</section>
<section id="methods" class="level2">
<h2 class="anchored" data-anchor-id="methods">Methods</h2>
<section id="sequence-and-disorder-data" class="level3">
<h3 class="anchored" data-anchor-id="sequence-and-disorder-data">Sequence and disorder data</h3>
<p>The human p53 amino acid sequence was obtained from UniProt (<a href="https://www.uniprot.org/uniprotkb/P04637/entry">P04637</a>). Individual residual disorder scores were obtained from a precomputed disorder prediction source (<a href="https://iupred2a.elte.hu/">IUPred2A</a>). Disorder scores were matched to the p53 sequence by residue number.</p>
</section>
<section id="domain-annotation" class="level3">
<h3 class="anchored" data-anchor-id="domain-annotation">Domain annotation</h3>
<p>Functional domains of p53 were defined using residue ranges from scientifically reputable <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC3905691/?utm_source=chatgpt.com">sources</a>. Residues outside these regions were labeled as N/A.</p>
</section>
<section id="feature-calculation" class="level3">
<h3 class="anchored" data-anchor-id="feature-calculation">Feature calculation</h3>
<p>Kyte–Doolittle hydrophobicity, along with Individual residue charge values were assigned to each amino acid residue and smoothed using a rolling window average. A rolling mean average with a center alignment was utilized to simulate the idea that protein functionality is based on multi-residue sequential regions rather than individual residues. Median hydrophobicity values were calculated for each domain. Charge-related features were calculated directly from residue identity.</p>
</section>
<section id="data-analysis-and-visualization" class="level3">
<h3 class="anchored" data-anchor-id="data-analysis-and-visualization">Data analysis and visualization</h3>
<p>All analyses were performed in R using standard data manipulation and visualization libraries. Domain-level comparisons were visualized using line plots and boxplots.</p>
</section>
</section>
<section id="results" class="level1">
<h1><strong>Results</strong></h1>
<section id="intrinsic-disorder-profile-along-p53" class="level2">
<h2 class="anchored" data-anchor-id="intrinsic-disorder-profile-along-p53">Intrinsic disorder profile along p53</h2>
<p>Intrinsic disorder scores vary substantially along the p53 amino acid sequence. Elevated disorder is observed in the N-terminal domains and the C-terminal region, while the central portion of the protein exhibits singificant lower disorder scores. This psignificantlyattern indicates a clear separation between flexible regulatory regions and a structured core.</p>
</section>
<section id="domain-level-comparison-of-intrinsic-disorder" class="level2">
<h2 class="anchored" data-anchor-id="domain-level-comparison-of-intrinsic-disorder">Domain-level comparison of intrinsic disorder</h2>
<p>Comparison of disorder scores across annotated domains reveals strong domain-specific differences. The N-terminal domains (TAD1 and TAD2), the proline-rich region, and the C-terminal domain display high intrinsic disorder, consistent with associated roles in gene regulation. In contrast, the DNA-binding domain exhibits substantially lower disorder, indicating a predominantly structured region.</p>
</section>
<section id="hydrophobicity-by-domain" class="level2">
<h2 class="anchored" data-anchor-id="hydrophobicity-by-domain">Hydrophobicity by domain</h2>
<p>Median Kyte–Doolittle hydrophobicity values differed across p53 domains. The DNA-binding domain showed the least negative median value (−0.5), consistent with a structured and stably folded region. In contrast, regulatory and disordered regions exhibited more negative median values, including TAD1 (−0.75), TAD2 (−1.2), and the C-terminal domain (−1.3), indicating increased hydrophilicity. The oligomerization domain occupied an intermediate range (−0.8), while residues that did not belong to any regions displayed the most hydrophilic values (−1.6), suggesting their role in linkage capacity.</p>
</section>
</section>
<section id="discussion" class="level1">
<h1><strong>Discussion</strong></h1>
<p>This analysis demonstrates that intrinsic disorder in p53 is strongly dependent on domain organization. Regulatory regions at the N- and C-termini exhibit high disorder and hydrophilicity, congruent with their relevant roles in protein–protein interactions and post-translational regulation. In contrast, the DNA-binding domain stands relatively structured and comparatively less hydrophilic, a reflection of the importance of stable folding in the context of sequence-specific binding.</p>
<p>Hydrophobicity patterns closely mirror disorder distributions, reinforcing the distinction between structured and intrinsically disordered regions. These differences among different domains remain pronounced despite being smoothed over via rolling averages, suggesting that they hard coded.</p>
<p>Overall, these results illustrate how simple sequence-derived features can capture the functional architecture of p53, separating its structured core from flexible regulatory regions without the need for complex modeling or experimental data.</p>
</section>
<section id="limitations" class="level1">
<h1><strong>Limitations</strong></h1>
<p>This study focuses on a single protein relying entirely on computational predictions and sequence-based features. No experimental validation was performed, and disorder predictions may vary depending on the algorithm/source used. Consequently, the findings should be interpreted as descriptive rather than definitive.</p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>
